pootle/apps/pootle_store/views.py

```diff
@@ -519,8 +523,24 @@ def submit(request, unit):
     form = form_class(request.POST, instance=unit, request=request)
 
     if form.is_valid():
+        suggestion = form.cleaned_data['suggestion']
+        if suggestion:
+            old_unit.accept_suggestion(suggestion,
+                                       request.translation_project, request.user)
+            if form.cleaned_data['comment']:
+                kwargs = dict(
+                    comment=form.cleaned_data['comment'],
+                    user=request.user,
+                )
+                comment_form = UnsecuredCommentForm(suggestion, kwargs)
+                if comment_form.is_valid():
+                    comment_form.save()
+
         if form.updated_fields:
             for field, old_value, new_value in form.updated_fields:
+                if (field == SubmissionFields.TARGET
+                    and form.cleaned_data['suggestion']):
```

Comment:
He usually calculates the conditions and stores the result in a variable, then use the variable in the `if`.